### ISSUE TITLE
failed to load plugin

### ISSUE THREAD

**Issue by cristianmtr** [obsidian-html/obs.html#1]
Hello,

I am trying to use this plugin. However, when I try to activate I get an error "Failed to load plugin" and it doesn't activate.

0.14.15 on Ubuntu

**Comment by hgrw** [comment]
Not sure if its the official way to do it but I got it to work by installing with npm. 

`npm run build`. If main.js is generated, it will work. You'll have to install typescript too.

**Comment by gpQ4utjefs** [comment]
> Hello,
> 
> I am trying to use this plugin. However, when I try to activate I get an error "Failed to load plugin" and it doesn't activate.
> 
> 0.14.15 on Ubuntu

Same problem

**Comment by dwrolvink** [comment]
> Not sure if its the official way to do it but I got it to work by installing with npm.
> 
> `npm run build`. If main.js is generated, it will work. You'll have to install typescript too.

Yes this is a step that I use when developing it, I assumed that once the main.js is generated this would not be necessary.

@cristianmtr apologies for the late reaction, I don't seem to get emails for this project for some reason.
Does the npm run build work for you? This is really still in alpha mode related to this [item](https://github.com/obsidian-html/obsidian-html/issues/271), I can look this weekend whether I can get it to work without any npm dependencies.

**Comment by sa-aguilarv** [comment]
> Hello,
> 
> I am trying to use this plugin. However, when I try to activate I get an error "Failed to load plugin" and it doesn't activate.
> 
> 0.14.15 on Ubuntu

Hey! I'm having the same problem. I tried to solve it by:

1. Installing npm: ``sudo apt install npm`` [source](https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-20-04)
2. Installing typescript: 
``curl -fsSL https://deb.nodesource.com/setup_15.x | sudo -E bash -``
``sudo apt-get install -y nodejs``
``sudo npm install -g typescript``
``tsc``
[source](https://dev.to/quentindamianino/how-to-install-typescript-on-ubuntu-33am)

After that, I did the ``npm run build`` and among the errors (see errors1.txt) it said to do ``npm i --save-dev @types/node`` which I was unable to install until I did:
1. ``cd`` to get out of the obs-html directory
2. ``sudo npm i --save-dev @types/node``

Once again I got multiple errors when I tried to install the plugin with npm (see errors2.txt). I realized there were a lot of missing modules/properties/... so I did the following:
1. ``npm i obsidian`` [source](https://www.npmjs.com/package/obsidian)
2. ``npm i require`` [source](https://www.npmjs.com/package/require)
3. ``npm i tslib`` [source](https://www.npmjs.com/package/tslib)

Here I stopped because I might be getting this all wrong. I don't have experience with npm, so I would be grateful if someone could help me out.

Thank you in advance for reading my message.

[errors1.txt](https://github.com/obsidian-html/obs.html/files/9816739/errors1.txt)
[errors2.txt](https://github.com/obsidian-html/obs.html/files/9816770/errors2.txt)

**Comment by dwrolvink** [comment]
All that I did, to install and use the plugin is this: 
``` bash
npm i          # to install dependencies
npm run dev    # to start compilation in watch mode
```
See also https://github.com/obsidianmd/obsidian-sample-plugin#how-to-use

Though after compiling I can just reboot my pc and have the plugin work, so I thought the plugin should just work as-is, when the main.js is created, but this is apparently not the case...

Will have to look into this further, I'm not a nodejs guru.

**Comment by sa-aguilarv** [comment]
> All that I did, to install and use the plugin is this:
> 
> ```shell
> npm i          # to install dependencies
> npm run dev    # to start compilation in watch mode
> ```
> 
> See also https://github.com/obsidianmd/obsidian-sample-plugin#how-to-use
> 
> Though after compiling I can just reboot my pc and have the plugin work, so I thought the plugin should just work as-is, when the main.js is created, but this is apparently not the case...
> 
> Will have to look into this further, I'm not a nodejs guru.

I'm sorry, I just found the error. I was running everything on my HDD instead of my SSD. I normally work with virtual environments in the HDD so it took me a while to notice. Once I gave it a try and did what you mentioned, it worked. Thank you again for helping me out!

**Comment by welpdx** [comment]
to anyone who had this [error](https://i.imgur.com/wc66KDg.png) when trying to get dataview working for obsidian-html:

this is what worked for me:
```bash
cd /path/to/vault/.obsidian/plugin/
git clone https://github.com/obsidian-html/obs.html
cd obs.html
sudo apt install npm
sudo apt install -y nodejs
sudo npm install -g typescript
npm i
npm run dev
```

in obsidian:
now reload plugins
enable plugins
